src/vs/workbench/services/preferences/common/preferencesModels.ts

```diff
@@ -909,8 +909,7 @@ class SettingsContentBuilder {
 
 		setting.descriptionRanges = [];
 		const descriptionPreValue = indent + '// ';
-		for (let line of setting.description) {
-			// Remove setting link tag
+		for (let line of (setting.deprecationMessage ? [setting.deprecationMessage] : setting.description)) {
```

Comment:
It only writes the deprecation message instead of the description?

Reply:
Could make it write both, but the concern from earlier was that it was not totally clear that the item was deprecated. Maybe it would be better with the deprecation message at the top? What do you think?

Reply:
I think it should have both. How about just prepending the deprecationMessage in `*` so it gets rendered in italics? And we can decide whether we want something fancier later.

Reply:
As discussed, this is the raw text model so we dont have italic support. In the new editor we use the special deprecation warning element anyways